### DISCUSSION TITLE
How to delete a user ?

### DISCUSSION THREAD

**Discussion by KaKi87** [tolgee/tolgee-platform#1026] [Q&A]
Either as a user or as the instance owner ?

Thanks

**Comment by JanCizmar** [comment]
EDIT: User can be deleted since 2.45.0 in their user profile..

OLD ANSWER:
It's currently not supported via UI or API. It's part of this issue I just created. 
https://github.com/tolgee/tolgee-platform/issues/1031

**Comment by fernandodevelon** [comment]
I am trying with docker latest version.
delete user. delete organisation.
but it does not work!

**Reply by JanCizmar** [reply]
Can you provide more information?

**Reply by fernandodevelon** [reply]
All right. this is the sequence of operations I have done.

1. invite new user into project
2. login and logout with new user
3. admin delete user from organization
4. the user is still present and an organization is created with his name

**Reply by JanCizmar** [reply]
https://github.com/tolgee/tolgee-platform/issues/1325 I am going to fix it this week. Sorry for inconvenience.

**Comment by JanCizmar** [comment]
Fixed here: https://github.com/tolgee/tolgee-platform/pull/1377

**Reply by KaKi87** [reply]
Thank you, could you please notify us here, once it's released ?

**Reply by JanCizmar** [reply]
It's reelased. You can also subscube for updates on the PR, next time 👍🏻

**Reply by KaKi87** [reply]
Oh, I didn't realized there was a bot that comments when it's released. (On most repos, subscribing to PRs is useless since you get notified when it's merged but not when it's released). Thanks

**Reply by JanCizmar** [reply]
No problem. 🍭

**Reply by KaKi87** [reply]
So, I installed the latest version, and didn't find a deletion button on the user, so I guess I still need to use the organization deletion button, which gives successfull confirmation and redirects to home (which is honestly annoying) but when getting back to the organization list, it's still there. Thanks

**Reply by JanCizmar** [reply]
True, there is no delete user button in the organization, but users can delete their accounts. 
As a workaround you can do it by debugging user account and deleting via their profile. 

This should be fixed. Sorry.

**Reply by KaKi87** [reply]
> there is no delete user button in the organization

Actually I was taking about `/administration/users`.

On the contrary, a user deletion button in the organization could mislead users to believe that it would remove the user from the organization (as does the current `X` button), therefore unexpectedly delete a user from the whole platform.
> As a workaround you can do it by debugging user account and deleting via their profile.

That works, although it terminates my admin session so I have to log back in each time.
> This should be fixed.

Please notify me once there's a PR for that.

Also, what do I do about "No auth user | ___implicit_user" ?

Thanks

**Reply by JanCizmar** [reply]
> Actually I was taking about /administration/users. 

I meant administration, not organization. My bad.

> Also, what do I do about "No auth user | ___implicit_user" ?

You can delete it when it's fixed. This user is created when authentication is disabled. You've probably enabled authentication later and that's why the user was created.

**Reply by JanCizmar** [reply]
There is the PR. Just need to test it. https://github.com/tolgee/tolgee-platform/pull/1392

**Reply by KaKi87** [reply]
That works !

One last thing :

![Screenshot_20221008-011904_Kiwi Browser-01](https://user-images.githubusercontent.com/21284089/194674595-d758fa32-ea5b-49b5-8efa-3e0c715004d8.jpeg)

Thanks
